chapter-4/ch4-cloudwatchlog-alarm.yml

```diff
@@ -0,0 +1,204 @@
+      - Label:
+          default: CloudWatch Log Group Requirements
+        Parameters:
+          - LogGroupName
+          - LogGroupRetention
+      - Label:
+          default: CloudWatch Log Metric Filter Requirements
+        Parameters:
+          - LogFilterMetricValue
+          - LogFilterMetricNamespace
+          - LogFilterMetricName
+          - LogFilterPattern
+      - Label:
+          default: CloudWatch Alarm Requirements
+        Parameters:
+          - LogAlrPeriod
+          - LogAlrThreshold
+          - SNSTopicForCWAlarms
+      - Label:
+          default: CloudWatch Anomaly Detector Requirements
+        Parameters:
+          - AnomalyStat
+          - AnomalyDimName
+          - AnomalyDimValue
+          - AnomalyNameSpace
+          - AnomalyMetricName
+          - AnomalyConfigZone
+          - AnomalyConfigExcludeEnd
+          - AnomalyConfigExcludeStart
+
+#==================================================
+# Parameters
+#==================================================
+
+Parameters:
+  LogFilterPattern:
+    Type: String
+    Description: 'cloudwatch log metric filter'
+    Default: "[ip, identity, user_id, timestamp, request, status_code = 4*, size]"
+  LogFilterMetricValue:
+    Type: String
+    Description: 'cloudwatch log metric filter value'
+    Default: "1"
+  LogFilterMetricNamespace:
+    Type: String
+    Description: 'cloudwatch log metric filter name space'
+    Default: "WebServer/400s"
+  LogFilterMetricName:
+    Type: String
+    Description: 'cloudwatch log metric filter name'
+    Default: "4xxCount"
+  SNSTopicForCWAlarms:
+    Type: String
+    Description: Provide SNS Topic ARN for standard CloudWatch alarms.
+    Default: arn:aws:sns:us-west-2:123456789012:SNS
+  LogAlrPeriod:
+    Type: Number
+    Default: '600'
+    Description: The time over which the specified statistic is applied. Specify time in seconds, in multiples of 60.
+  LogAlrThreshold:
+    Type: Number
+    Default: '90'
+    Description: The count threshold
+  AnomalyMetricName:
+    Type: String
+    Description: 'cloudwatch anomaly metric name'
+    Default: "JvmMetric"
+  AnomalyNameSpace:
+    Type: String
+    Description: 'cloudwatch anomaly name space'
+    Default: "AWSSDK/Java"
+  AnomalyStat:
+    Type: String
+    Description: 'cloudwatch anomaly stat'
+    Default: "Average"
+  AnomalyDimName:
+    Type: String
+    Description: 'cloudwatch anomaly dimension name'
+    Default: "Memory"
+  AnomalyDimValue:
+    Type: String
+    Description: 'cloudwatch anomaly dimension value'
+    Default: "UsedMemory"
+  AnomalyConfigZone:
+    Type: String
+    Description: 'cloudwatch anomaly time zone'
+    Default: "UTC"
+  AnomalyConfigExcludeStart:
+    Type: String
+    Description: 'cloudwatch anomaly exclude start time'
+    Default: "2022-07-01T00:00:00"
+  AnomalyConfigExcludeEnd:
+    Type: String
+    Description: 'cloudwatch anomaly exclude end time'
+    Default: "2022-07-01T23:59:59"
```

Comment:
if exclude is same as include will it show the right data?

Reply:
The value is different to define a time range